tests/tests_limit_allocator.cpp

```diff
@@ -0,0 +1,39 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+
+#include <dice/template-library/limit_allocator.hpp>
+
+TEST_SUITE("limit_allocator sanity check") {
+	using namespace dice::template_library;
+
+	template<limit_allocator_syncness syncness>
+	void run_tests() {
+		limit_allocator<int> alloc{12 * sizeof(int)};
+
+		auto a = alloc.allocate(5);
+		auto b = alloc.allocate(5);
+		auto c = alloc.allocate(1);
+		alloc.deallocate(c, 1);
+		alloc.deallocate(b, 5);
+
+		CHECK_THROWS(alloc.allocate(8));
+
+		auto d = alloc.allocate(7);
+
+		alloc.deallocate(a, 5);
+		alloc.deallocate(d, 7);
+
+		auto e = alloc.allocate(12);
+		alloc.deallocate(e, 12);
+
+		CHECK_THROWS(alloc.allocate(13));
+	}
+
+	TEST_CASE("sync") {
+		run_tests<limit_allocator_syncness::sync>();
```

Comment:
This does not really test the syncedness. But we also do not really plan to use it synced for now. Maybe move to an issue and resolve later?

Reply:
yes, I'll create an issue. Does the sync look fine to you?

Reply:
yes, it does